shell/platform/android/io/flutter/embedding/FlutterView.java

```diff
@@ -38,1015 +33,781 @@
+      }
+      flutterRenderer.setSemanticsEnabled(false);
+    }
+    resetWillNotDraw();
+  }
+
+  @Override
+  public AccessibilityNodeProvider getAccessibilityNodeProvider() {
+    if (mAccessibilityEnabled)
+      return mAccessibilityNodeProvider;
+    // TODO(goderbauer): when a11y is off this should return a one-off snapshot of
+    // the a11y
+    // tree.
+    return null;
+  }
+
+  private void resetWillNotDraw() {
+    if (!mIsSoftwareRenderingEnabled) {
+      setWillNotDraw(!(mAccessibilityEnabled || mTouchExplorationEnabled));
+    } else {
+      setWillNotDraw(false);
+    }
+  }
+  //----- END AccessibilityStateChangeListener ----
+
+  //------- START ACCESSIBILITY ------
+  public void dispatchSemanticsAction(int id, AccessibilityBridge.Action action) {
+    dispatchSemanticsAction(id, action, null);
+  }
+
+  public void dispatchSemanticsAction(int id, AccessibilityBridge.Action action, Object args) {
+    if (!isAttachedToRenderer)
+      return;
+    ByteBuffer encodedArgs = null;
+    int position = 0;
+    if (args != null) {
+      encodedArgs = StandardMessageCodec.INSTANCE.encodeMessage(args);
+      position = encodedArgs.position();
+    }
+    flutterRenderer.dispatchSemanticsAction(id, action.value, encodedArgs, position);
+  }
+
+  public void updateAccessibilityFeatures() {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+      String transitionAnimationScale = Settings.Global.getString(getContext().getContentResolver(),
+          Settings.Global.TRANSITION_ANIMATION_SCALE);
+      if (transitionAnimationScale != null && transitionAnimationScale.equals("0")) {
+        mAccessibilityFeatureFlags ^= AccessibilityFeature.DISABLE_ANIMATIONS.value;
+      } else {
+        mAccessibilityFeatureFlags &= ~AccessibilityFeature.DISABLE_ANIMATIONS.value;
+      }
+    }
+    flutterRenderer.setAccessibilityFeatures(mAccessibilityFeatureFlags);
+  }
+
+  void ensureAccessibilityEnabled() {
+    if (!isAttachedToRenderer)
+      return;
+    mAccessibilityEnabled = true;
+    if (mAccessibilityNodeProvider == null) {
+      mAccessibilityNodeProvider = new AccessibilityBridge(this, pluginMessenger);
+    }
+    flutterRenderer.setSemanticsEnabled(true);
+    mAccessibilityNodeProvider.setAccessibilityEnabled(true);
+  }
+
+  void resetAccessibilityTree() {
+    if (mAccessibilityNodeProvider != null) {
+      mAccessibilityNodeProvider.reset();
+    }
+  }
+  //------- END ACCESSIBILITY ----
+
+  //----- START FLUTTER INTEGRATION -----
+
+  /**
+   * Start rendering the UI for the given {@link FlutterRenderer}.
+   *
+   * @param flutterRenderer the FlutterRenderer for which this FlutterView will be a RenderSurface
+   * @param pluginMessenger the BinaryMessenger to use with any plugins that this FlutterView needs
+   *                        to register
+   */
+  public void attachToFlutterRenderer(@NonNull FlutterRenderer flutterRenderer, @NonNull BinaryMessenger pluginMessenger) {
```

Comment:
I'm curious, when we're running dart code in the background do we still need to be attached to a renderer? (is it guaranteed that a FlutterView exist at all?) if not, what happens with message channels?

Reply:
Some of those details are yet to be specified, but I intend for background execution to not require a FlutterView at all, because to do would make little sense.

If I'm not mistake, message channels are registered with a BinaryMessenger which is implemented by FlutterEngine, not FlutterView.